src/Integration.Vsix.UnitTests/SonarLintTagger/TaggerProviderTests.cs

```diff
@@ -127,7 +128,43 @@ public void CreateTagger_should_return_null_when_analysis_is_not_supported()
         }
 
         [TestMethod]
-        public void CreateTagger_should_return_same_tagger_for_already_tracked_file()
+        public void CreateTagger_SameDocument_ShouldUseSameSingletonManager()
+        {
+            var doc1 = CreateMockedDocument("doc1.js");
+            var buffer = doc1.TextBuffer;
+
+            CheckPresenceOfSingletonManagerInPropertyCollection(buffer, false);
+
+            // 1. Create first tagger for doc
+            var tagger1 = CreateTaggerForDocument(doc1);
+            var firstRequestManger = CheckPresenceOfSingletonManagerInPropertyCollection(buffer, true);
+
+            // 2. Create second tagger for doc
+            var tagger2 = CreateTaggerForDocument(doc1);
+            var secondRequestManager = CheckPresenceOfSingletonManagerInPropertyCollection(buffer, true);
+
+            firstRequestManger.Should().BeSameAs(secondRequestManager);
+        }
+
+        [TestMethod]
+        public void CreateTagger_DifferentDocuments_ShouldUseDifferentSingletonManagers()
+        {
+            var doc1 = CreateMockedDocument("doc1.js");
+            var doc2 = CreateMockedDocument("doc2.js");
+
+            // 1. Create tagger for first doc
+            var tagger1 = CreateTaggerForDocument(doc1);
+            var doc1Manager = CheckPresenceOfSingletonManagerInPropertyCollection(doc1.TextBuffer, true);
+
+            // 2. Create tagger for second doc
+            var tagger2 = CreateTaggerForDocument(doc2);
+            var doc2Manager = CheckPresenceOfSingletonManagerInPropertyCollection(doc2.TextBuffer, true);
+
+            doc1Manager.Should().NotBeSameAs(doc2Manager);
+        }
+
+        [TestMethod]
+        public void CreateTagger_should_return_new_tagger_for_already_tracked_file()
```

Comment:
This test and the next one are pre-existing tests that I've modified.
In fact, I've reverted both tests to the form they were in in commit 721e343a on 4th September 2020.